crypt.go

```diff
@@ -271,6 +398,10 @@ func activateWithRecoveryKey(volumeName, sourceDevicePath string, keyReader io.R
 			continue
 		}
 
+		if err := keyring.AddKeyToUserKeyring(key[:], sourceDevicePath, keyringPurposeDiskUnlock, keyringPrefixOrDefault(keyringPrefix)); err != nil {
```

Comment:
the older code would distinguish at this level whether we had the main key or a recovery key, but now it seems we use "unlock" for either?

Reply:
It doesn't matter now. The previous way also tried passing in some error status via the key description, which was a bit of a mess. A future PR will add a mechanism to serialize the unlocking state so that the system can determine if the key used to unlock it was a recovery key, and then use that key to create recreate the primary key(s).